### PR TITLE
Add object info query

### PR DESCRIPTION
**Pull request by progschj**
see #25 and #24
Also includes the `ANARI_STRING_LIST`, `ANARI_TYPE_LIST` and `ANARI_PARAMETER_LIST`. Mentioned in #21.

### CODE REVIEW

File: chapters/type_table.txt [hunk 1]
```diff
@@ -8,6 +8,18 @@
 |`const char*`
 | 0-terminated string
 
+|`TYPE_LIST`
```
Comment (johguenther) [inline chapters/type_table.txt:11]: Shouldn't this be `DATA_TYPE_LIST`?

File: chapters/type_table.txt [hunk 2]
```diff
@@ -8,6 +8,18 @@
...
+|`ANARIDataType*`
+| array of `ANARIDataType` values terminated by `ANARI_UNKNOWN`
```
Comment (johguenther) [inline chapters/type_table.txt:13]: In tables we drop the `ANARI_` prefix of datatypes constants:
> array of `ANARIDataType` values terminated by `UNKNOWN`

File: chapters/api_concepts.txt [hunk 3]
```diff
@@ -247,18 +247,31 @@ typedef struct
...
-An object (sub)type `objectType`/`objectSubtype` can be queried for supported
-parameters with `anariGetObjectParameters` (passing `NULL` or an empty string
+An object (sub)type `objectType`/`objectSubtype` can be queried for information
+with `anariGetObjectInfo` (passing `NULL` or an empty string
 for `objectSubtype` to query an object type directly that does not have
-subtypes). The function returns `NULL` if there are no parameters, or an array
-of `ANARIParameter` structs, where the last element has `NULL` as name and
-`ANARI_UNKNOWN` as type.
+subtypes). The function returns the result for `infoName` of type `infoType`, or
+`NULL` if the information can not be retrieved.
```
Comment (johguenther) [inline chapters/api_concepts.txt:262]: Can we change to the more formal "cannot" (can not) and "information" (info) in running text?

**Review by johguenther — CHANGES_REQUESTED** [review]

**Comment by progschj** [comment]
The last commit should address all the change requests.

**Pull request merged**